packages/evm/src/eof/verify.ts

```diff
@@ -0,0 +1,519 @@
+    while (ptr < code.length) {
+      // This set tracks the successor opcodes of this opcode (for stack purposes)
+      const successorSet = new Set<number>()
+
+      // ReachableOpcodes: this can likely be deleted after implementing the 5450 algorithm
+      if (!reachableOpcodes.has(ptr)) {
+        validationError(EOFError.UnreachableCode)
+      }
+
+      if (stackHeightMin[ptr] === undefined || stackHeightMax[ptr] === undefined) {
+        // This error either means that the code is unreachable,
+        // or it is possible that it is only reachable via a backwards jump
+        validationError(EOFError.UnreachableCode)
+      }
+
+      validJumps.add(ptr)
+      const opcode = code[ptr]
+
+      const minStackCurrent = stackHeightMin[ptr]
+      const maxStackCurrent = stackHeightMax[ptr]
+
+      const opcodeInputs = stackDelta[opcode].inputs
+      const opcodeOutputs = stackDelta[opcode].outputs
+
+      if (minStackCurrent - opcodeInputs < 0) {
+        validationError(EOFError.StackUnderflow)
+      }
+
+      const delta = opcodeOutputs - opcodeInputs
+
+      let minStackNext = minStackCurrent + delta
+      let maxStackNext = maxStackCurrent + delta
+
+      if (maxStackNext > 1023) {
+        // TODO verify if 1023 or 1024 is the right constant
+        validationError(EOFError.StackOverflow)
+      }
+
+      if (returningFunction && opcode === 0xe4) {
+        validationError(EOFError.InvalidReturningSection)
+      }
+
+      lastOpcode = opcode
+      if (!opcodeNumbers.has(opcode)) {
+        validationError(EOFError.InvalidOpcode)
+      }
+
+      if (opcode === 0xe0 || opcode === 0xe1) {
+        // RJUMP / RJUMPI
+        const target = readInt16(code, ptr + 1) + ptr + 3
+        if (target < 0 || target >= code.length) {
+          validationError(EOFError.InvalidRJUMP)
+        }
+
+        successorSet.add(target)
+
+        addJump(target)
+        reachableOpcodes.add(target)
+
+        if (opcode === 0xe0) {
+          // For RJUMP check that the instruction after RJUMP is reachable
+          // If this is not the case, then it is not yet targeted by a forward jump
+          // And hence violates the spec
+          if (!reachableOpcodes.has(ptr + 3) && ptr + 3 < code.length) {
+            // Note: the final condition above ensures that the bytes after ptr are there
+            // This is an edge case, if the container ends with RJUMP (which is valid)
+            validationError(EOFError.UnreachableCode)
+          }
+        }
+      } else if (opcode === 0xe2) {
+        // RJUMPV
+        const tableSize = code[ptr + 1] + 1
+
+        if (tableSize === undefined) {
+          validationError(EOFError.OpcodeIntermediatesOOB)
+        } else if (tableSize === 0) {
+          validationError(EOFError.RJUMPVTableSize0)
+        }
+
+        if (ptr + tableSize * 2 + 2 >= code.length) {
+          // Fall-through case
```

Comment:
Can we explain what the fall-through case is?  If I understand the spec correctly, this is where the `case` in the stack, is greater than the highest possible index to jump to so it just skips over the intermediates to the next opcode, right?

Reply:
No, the problem here is that RJUMPV is either the final opcode (this is not allowed) or, if it would "fall through" (the `case` on stack is not within the table indices) the program counter would end up out-of-bounds i.e. outside the code. Sorry, this comment is confusing.

General note: in `verify.ts` we do not have access to any stack, so we cannot reason about it, and thus have to consider all values